simnode/generator.py

```diff
@@ -85,21 +94,137 @@ def _calculate_parallel_coords(offset: float, line_width: float) \
                     marking_widths = repeat(line_width, num_coords)
                     return list(zip(coords[0], coords[1], z_vals, marking_widths))
 
+                def _calculate_offset_list(relative_offset: float, absolute_offset: float,
+                                           output_number_of_points: int = self.add_roads_to_scenario.num_nodes//3) \
+                                            -> List[float]:
+                    """ calculates a list of relative offsets to the road centre
+                    uses new_width_vals for dynamic offset
+                    change the default value of output_number_of_points for more precision, has to be less \
+                    than num_nodes//2
+
+                    :param relative_offset: relative to the width of the road, must be between -0.5 and 0.5
+                    :param absolute_offset: absolute, to account for line width
+                    :param output_number_of_points: number of outputs in list
+                    :return: list of width offsets
+                    """
+                    assert 0 < output_number_of_points < new_width_vals.__len__()//2, \
+                        "choose a valid number of output vals"
+                    assert -0.5 <= relative_offset <= 0.5, "relative offset must be between -0.5 and 0.5"
+                    assert -max(new_width_vals)/2 < absolute_offset < max(new_width_vals)/2, \
+                        "absolute offset must be smaller than half of the road"
+                    cutting_points = linspace(0, new_width_vals.__len__()-1, dtype=int, num=output_number_of_points)
+                    output_vals = list(map(lambda i: new_width_vals[i]*relative_offset + absolute_offset, cutting_points))
+                    return output_vals
+
+                def _calculate_parallel_pieces(offset: List[float], original_line: LineString, cutting_points: ndarray)\
+                                                -> Tuple[array, array]:
+                    # TODO how to specify the type of an array? Should be float...
+                    """ This method will calculate offsets for smaller pieces of road.
+
+                    :param offset: list of width offsets, must be smaller than num_nodes//2, should be equidistant
+                    :param original_line: LineString of baseline road coordinates
+                    :param cutting_points: array of points at which the road is split into pieces
+                    :return: Returns a tuple of float arrays for x and y values
+                    """
+                    from shapely.errors import TopologicalError
+
+                    offset_sub_lines = ()
```

Comment:
Why is it a tuple? I don't see where you use it without accessing its by indices. If you split this tuple into two separate variables you could even clarify their meaning by given appropriate names.

Reply:
It is a tupel because that is the return type of shapely. Will change.